src/clr-addons/daterangepicker/directives/daterangepicker/daterangepicker.directive.ts

```diff
@@ -0,0 +1,306 @@
+import {
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+  Renderer2,
+  Self,
+} from '@angular/core';
+import { ControlValueAccessor, NgControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+
+import { SEPARATOR_TEXT_DEFAULT } from '../../daterange.constants';
+import { NullableDaterange } from '../../interfaces/daterange.interface';
+import { DayModel } from '../../models/day.model';
+import { DaterangeService } from '../../providers/daterange.service';
+import { DaterangeParsingService } from '../../providers/daterange-parsing.service';
+import { DaterangeControlStateService } from '../../providers/daterange-control-state.service';
+import { ControlIdService } from '../../../abstract-form-component/control-id.service';
+
+/**
+ * Daterangepicker.
+ */
+@Directive({
+  selector: '[clrDaterangepicker]',
+})
+export class ClrDaterangepickerDirective implements OnInit, OnDestroy, ControlValueAccessor {
+  /**
+   * Minimum date that can be selected.
+   */
+  @Input()
+  public set min(value: Date | undefined | null) {
+    this.daterangeService.minDate = value == null ? null : new DayModel(value);
```

Comment:
Checking with `== null` checks both `null` & `undefined`.
Only `=== null` does **not** check for `undefined`.
I always stay away from using the bang, because lots of values are falsy.